### ISSUE TITLE
`GlobalFlag.getAll()` fails to validate flags defined within a package object.

### ISSUE THREAD

**Issue by caplan** [twitter/util#212]
`GlobalFlag.getAll()` fails to validate flags defined within a package object.

This seems to be caused by improper parsing of the `$` delimiters in class names - a global flag `flag` within a package object `packageObject` is represented as `packageObject.package$flag$`, while the `GlobalFlag.getAll()` assumes `packageObject.package$flag` (no trailing `$`).

I can go ahead and submit a pr for this if that's ok with you guys.

### Expected behavior

when declaring a global flag inside a package object, like so: 

```scala
package object funStuff {
  object enableMoreFun extends GlobalFlag[Boolean](true, "Enable More fun")
}
```

, the result of `GlobalFlag.getAll()` should contain the `enableMoreFun` flag.

### Actual behavior

a message is printed to stdout: `failed for funStuff`, and the result of `GlobalFlag.getAll()`does not contain the `enableMoreFun` flag.

### Steps to reproduce the behavior

declare a global flag object within a package object, then call `GlobalFlag.getAll()`

**Comment by kevinoliver** [comment]
Thanks. This is partially why we suggest defining global flags outside of any object or package. That said, I think we should fix this and a PR would be great.

**Comment by monkeyroar** [comment]
This is resolved in #242
